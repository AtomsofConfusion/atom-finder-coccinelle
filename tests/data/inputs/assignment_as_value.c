#include <stdio.h>

int add(int a, int b);

int main() {
    int v1, v2, v3, v4;
    v1 = 5;
    v2 = 7;
    v3 = 10;

    v1 = v2 = v3 + 5;

    v1 = v2 = v3 = 5;

    v1 = v2 = v3 = v4 = 10;

    v1 += v2 -= v3;

    printf("v1: %d, v2: %d\n", v1, v2);

    int v5 = v1 = 6;

    int v6 = v2 = v3 = v4 = 7;

    int v7 = v1 == v2;

    int v8;
    v8 |= v6 = add(v1, v2);

    int v9;
    v9 = v1 += (v2 -= 4);

    int v10 = (v1 += v2);
    int v11 = v1 += (v2 -= v3);
    int v12 = (v1 += (v2 -= v3));
    int v13;
    v12 = v3 + (v1 -= 2);
    v1 = v2 += v3 - (v4 += 2);
    v1 = (v2 += v3 - (v4 += 2));
    v1 = v2 + (v3 -= v4 += 2);
    v1 = v2 + v3 + (v4 += v5);

    for (int i = 0; v1 = v2 = i < 5; v3 = v4 -= 2, i++) {
        printf("%d\n", i);
    }

    for (int i = v5 = 0; i < 5; i++) {
        printf("%d\n", i);
    }

    for (int i = (v1 += v3 = 2); i < 10; i++) {
        printf("%d\n", i);
    }

    while (v1 < 5, v3 += v1) {
        v1++;
    }

    while (add(1, 3), v1 = v3 += v4 - 1, v1 > 0 && v1 < 10) {
        printf("%d\n", v1);
    }

    while (v1 += v2 = 2) {
        if (v1 >= 20) {
            break;
        }
    }

    if (v1 < 10, v1 += 5) {
        v2 -= 2;
    }

    if (v1 < 10, v2 += 5) {
        v3 = v4 -= v6 = 2;
    } else if (v2 < 10, v1 += 5) {
        printf("%d\n", v2);
    } else if (v3 < 10, v4 = v3) {
        printf("%d\n", v4);
    }

    return 0;
}

int add(int a, int b) {
    return a + b;
}