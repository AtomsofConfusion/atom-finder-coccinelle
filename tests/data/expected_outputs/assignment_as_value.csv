assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,33,4, int v10 = ( v1 += v2 ) ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,34,4, int v11 = v1 += ( v2 -= v3 ) ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,35,4, int v12 = ( v1 += ( v2 -= v3 ) ) ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,21,4, int v5 = v1 = 6 ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,23,4, int v6 = v2 = v3 = v4 = 7 ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,13,4,v1 = v2 = v3 = 5
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,15,4,v1 = v2 = v3 = v4 = 10
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,43,20,v1 = v2 = i < 5
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,11,4,v1 = v2 = v3 + 5
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,38,4,v1 = v2 += v3 - ( v4 += 2 )
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,59,22,v1 = v3 += v4 - 1
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,41,4,v1 = v2 + v3 + ( v4 += v5 )
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,39,4,v1 = ( v2 += v3 - ( v4 += 2 ) )
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,63,11,v1 += v2 = 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,17,4,v1 += v2 -= v3
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,51,18,v1 += v3 = 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,37,4,v12 = v3 + ( v1 -= 2 )
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,43,37,v3 = v4 -= 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,40,15,v3 -= v4 += 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,74,13,v4 -= v6 = 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,28,4,"v8 |= v6 = add ( v1 , v2 )"
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,31,4,v9 = v1 += ( v2 -= 4 )
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,47,9, int i = v5 = 0 ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,51,9, int i = ( v1 += v3 = 2 ) ;
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,43,37,v3 = v4 -= 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,55,19,v3 += v1
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,43,42,v4 -= 2
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,69,17,v1 += 5
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,75,24,v1 += 5
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,73,17,v2 += 5
assignment-as-value,/home/ubuntu/atom-finder-coccinelle/tests/data/inputs/assignment_as_value.c,77,24,v4 = v3

